core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -258,24 +259,34 @@ private void removePendingAcquisition(SessionProtocol desiredProtocol, PoolKey k
      * Attempts to acquire a {@link Channel} which is matched by the specified condition immediately.
      *
      * @return {@code null} is there's no match left in the pool and thus a new connection has to be
-     *         requested via {@link #acquireLater(SessionProtocol, PoolKey, ClientConnectionTimingsBuilder)}.
+     *         requested via {@link #acquireLater(SessionProtocol, SerializationFormat,
+     *         PoolKey, ClientConnectionTimingsBuilder)}.
      */
     @Nullable
-    PooledChannel acquireNow(SessionProtocol desiredProtocol, PoolKey key) {
+    @SuppressWarnings("checkstyle:FallThrough")
+    PooledChannel acquireNow(SessionProtocol desiredProtocol, SerializationFormat serializationFormat,
+                             PoolKey key) {
         PooledChannel ch;
         switch (desiredProtocol) {
             case HTTP:
                 ch = acquireNowExact(key, SessionProtocol.H2C);
-                if (ch == null) {
+                if (ch == null && !serializationFormat.requiresNewConnection(SessionProtocol.H1C)) {
```

Comment:
How about passing `serializationFormat` to `acquireNowExact` to make a decision for reuse?
```java
private PooledChannel acquireNowExact(PoolKey key, SessionProtocol protocol,
                                      SerializationFormat serializationFormat) {
  if (serializationFormat.requiresNewConnection(protocol)) {
    return null;
  }
}
```